app/controllers/user_controller.rb

```diff
@@ -0,0 +1,13 @@
+class UserController < ApplicationController
+  # Follows a user.
+  def follow
+    Follow.new(follower_id: current_user.id, followed_user_id: params[:id]).save
+    redirect_to root_url
+  end
+
+  # Unfollows a user.
+  def unfollow
+    Follow.where(follower_id: current_user.id, followed_user_id: params[:id])[0].destroy
```

Comment:
what is the reason to extract the first element of the array with [0]? Could it be done in another way?

Reply:
The reason is to send the first match of the array to destroy the follower.
It could be done by looking up the Follow ID and matching the IDs of the follower user and the followed user to remove it.